### PR TITLE
Add zlibpkg dependency to ag package

### PR DESCRIPTION
**Pull request by uberhacker**
- Designate some dependencies as build only

### CODE REVIEW

File: packages/ag.rb [hunk 1]
```diff
@@ -12,11 +12,12 @@ class Ag < Package
...
-  depends_on "pkgconfig"
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "pkgconfig" => :build
   depends_on "pcre"
   depends_on "xzutils"
+  depends_on "zlibpkg"
 
   def self.build
     system "autoreconf", "-fiv"
```
Comment (cstrouse) [inline packages/ag.rb]: Do you know why this is necessary? Usually packagers should not run autoreconf.

Comment (uberhacker) [reply packages/ag.rb]: I'm guessing because of the autoreconf line.  @vince, can you please provide more details?

Comment (cstrouse) [reply packages/ag.rb]: I was asking specifically about the autoreconf line. I may have marked the wrong line.

Comment (vince) [reply packages/ag.rb]: I largely took this recipe from the homebrew package. Could be it's not needed but I stopped fiddling when the package worked.

**Comment by cstrouse** [comment]
Works on Samsung Chromebook Plus (aarch64).

**Pull request merged**